src/Agent.Plugins/PipelineCache/SavePipelineCacheV0.cs

```diff
@@ -29,6 +29,18 @@ public class SavePipelineCacheV0 : PipelineCacheTaskPluginBase
             string salt,
             CancellationToken token)
         {
+            if(!context.Variables.ContainsKey("agent.jobstatus"))
+            {
+                return;
+            }
+
+            var JobStatus = context.Variables["agent.jobstatus"];
+            if(JobStatus != null && !JobStatus.Value.Equals("Succeeded"))
+            {
+                context.Output($"Exiting because the JobStatus is {JobStatus.Value}");
```

Comment:
JobStatus?.Value

Reply:
Ideally the status of the **Post-Job: Cache** step is "skipped" in the case where we don't try to save the cache. This will make it more obvious to the user when the operation is skipped.

There may be a better way to do this, but maybe something like:

```cs
context.Output($"##vso[task.complete result=skipped;]Skipping save because the job status is {JobStatus.Value}");
```

Reply:
Modifying "Skipping dave cache..." for  now.

Reply:
Setting the task.complete result to `skipped` caused a weird result in the UI, so just output a warning message for now. we should look at this again in the future.